modules/oe_newsroom_newsletter/tests/src/Functional/SubscribeMultipleNewslettersTest.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_newsroom_newsletter\Functional;
+
+use Drupal\Tests\BrowserTestBase;
+use Drupal\Tests\oe_newsroom_newsletter\Traits\OeNewsroomNewsletterTrait;
+use Drupal\user\Entity\Role;
+
+/**
+ * Test the subscription to multiple newsletters.
+ *
+ * @group oe_newsroom_newsletter
+ */
+class SubscribeMultipleNewslettersTest extends BrowserTestBase {
+
+  use OeNewsroomNewsletterTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'node',
+    'oe_newsroom_newsletter_mock',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $defaultTheme = 'stark';
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    $this->setApiPrivateKey();
+    $this->configureNewsroom();
+    $this->configureNewsletter();
+    $this->grantPermissions(Role::load(Role::ANONYMOUS_ID), [
+      'subscribe to newsletter',
+      'unsubscribe from newsletter',
+    ]);
+    $this->createNewsletterPages(TRUE);
+  }
+
+  /**
+   * Test if a user can subscribe to multiple newsletters.
+   *
+   * @group oe_newsroom_newsletter
+   */
+  public function testSubscribeMultipleNewsletters(): void {
+    $assertSession = $this->assertSession();
+    $session = $this->getSession();
+    $page = $session->getPage();
+
+    // Subscribe multiple Newsletters.
+    $this->drupalGet($this->subscribePath);
+    $assertSession->pageTextContains('Subscribe to newsletter');
+    $assertSession->pageTextContains('This is the introduction text.');
+    $page->fillField('Your e-mail', 'mail@example.com');
+    $assertSession->pageTextContains('Newsletter lists');
+    $assertSession->pageTextContains('Please select which newsletter list interests you.');
+    $page->checkField('By checking this box, I confirm that I want to register for this service, and I agree with the privacy statement');
+    $page->pressButton('Subscribe');
```

Comment:
We need to wait for ajax to finish.

Reply:
@piotrsmykaj I don't know if you made a test for this, but the form should work with and without AJAx, but it will be displayed slightly differently.

Reply:
The ajax is tested by the class NewsletterConfigurationTest.

Reply:
True, this is not the FunctionalJavascript one.